### ISSUE TITLE
Clip to DEM - enhancement request

### ISSUE THREAD

**Issue by Llaves** [minorua/Qgis2threejs#167]
The canvas extent is used to generate the surface on which the image is draped. If the DEM does not fill the canvas, the region beyond the DEM is treated as 0. This is a real problem if the DEM is well above sea level, such as this one with minimum elevation 2740M. 
![dem2](https://user-images.githubusercontent.com/6778818/54078204-75700200-4281-11e9-8ba8-5099d34c05bc.PNG)

Offsetting the elevation (using vertical shift)  doesn't help - it just moves the "apron" to negative values.
The solution requires the user to adjust the canvas so that it has the same aspect ratio as the DEM and then to scale the view so the DEM exactly fills the canvas. Setting Vertical shift to auto then gives this:
![dem1](https://user-images.githubusercontent.com/6778818/54078220-c8e25000-4281-11e9-9c91-f8e4fe4be0ae.PNG)

It would be nice to automatically trim to the DEM extent if it is less than the canvas extent.

**Comment by kannes** [comment]
Have you tried "Clip DEM with polygon layer"?

**Comment by Llaves** [comment]
Yes, I've tried that, but there are two admittedly minor issues:
1. You need to create a polygon layers with matching extend to have something to clip to.
2. At least in my experience you need to first create a polygon with matching extent, you then need to buffer that layer inward by 10-20 meters to get a clip that doesn't wrap the image around the sides.

**Comment by DigDigDig** [comment]
Another hacky thing you can do is to use the 'Raster>Fill nodata...' tool.  If you set a search distance it will interpolate to fill gaps in your DEM but also produce a buffer zone along the outside edge of your elevation source.  That keeps ThreeJS from rendering any no data points along the edge as 0.

![image](https://user-images.githubusercontent.com/17930616/54838013-11483780-4c9e-11e9-8b5d-fb77c0cbb8f1.png)

![image](https://user-images.githubusercontent.com/17930616/54837851-bf9fad00-4c9d-11e9-92fe-11cda31a6933.png)

**Comment by kannes** [comment]
In modern versions you can clip the whole scene to a layer's extent. This might fix this, if your layer is rectangular.

**Comment by Exebecce** [comment]
Hi,
After I solved the problem like Llaves, the problem is back with the 2.7.1 version of the plugin.
But now, I can't find the vertical shift setting to solve it.

This is not a NoData problem.

Has anyone solved this problem with the new version ?

**Comment by jfmoyen** [comment]
I have the same problem, I second the request ! It would be nice to be able to separate nodata from 0 ....
